grammar/parsing.js

```diff
@@ -206,12 +206,80 @@ class Parsing extends EmbeddedActionsParser {
             return result;
         });
 
+        $.RULE('multiArray', () => {
+            const multiArr = [[]];
+            let singleArr = null;
+
+            const updateRow = (multiArr, newArr) => {
+                if (multiArr[0] === null)
+                    multiArr.push([])
+                for(let newArrIndex = 0; newArrIndex < newArr.length; newArrIndex++) {
+                    let currVal = newArr[newArrIndex];
+                    if (!Array.isArray(currVal)) {
+                        multiArr[0].push(currVal);
+                        continue;
+                    }
+                    if (multiArr.length <= newArrIndex)
+                            multiArr.push([]);
```

Comment:
I think there's a bug here. This `multiArr.push([])` does not guarantee that `multiArr[newArrIndex]` will be defined on L225.  To do that, you'd need to do a loop:

```js
  while (multiArr.length <= newArrIndex) { 
    multiArr.push([]);
  }
```
***But*** I don't think that's what we ultimately want here either.

Take a case where the input `newArr` contains a lot of non-array values:

```js
  newArr = [0, 1, 2, 3, 4, [5, 6, 7, 8]];
```

What is the intended value of `multiArr` after calling this `updateRow` function? In my head, I'm calculating that `multiArr` would look like this when the code hits L222:

```js
  [[0, 1, 2, 3, 4]];
```

Then, after L223:

```js
  [[0, 1, 2, 3, 4], []];
```

The problem now is that `newArrIndex` will be set to `5`, but `multiArr` only has a length of `2`. So on L225, we try to `multiArr[5].push()`, but `multiArr[5]` is not yet defined, so an error is thrown. If we change that `if` to a `while`, we'll avoid the error, but end up with empty values in `multiArr`, e.g.:

`multiArr` after while loop:
```js
[[0, 1, 2, 3, 4], [], [], [], [], []];
```

Final `multiArr`:

```js
[[0, 1, 2, 3, 4], [], [], [], [], [5, 6, 7, 8]];
```

TLDR; I can't tell the exact intention of this function, but the current implementation appears to be prone to bugs/errors. If possible, see if this function can be extracted and write some tests to ensure it's working as intended.